pkg/registry/batch/job/strategy.go

```diff
@@ -379,6 +379,7 @@ func getStatusValidationOptions(newJob, oldJob *batch.Job) batchvalidation.JobSt
 		isUncountedTerminatedPodsChanged := !apiequality.Semantic.DeepEqual(oldJob.Status.UncountedTerminatedPods, newJob.Status.UncountedTerminatedPods)
 		isReadyChanged := !ptr.Equal(oldJob.Status.Ready, newJob.Status.Ready)
 		isTerminatingChanged := !ptr.Equal(oldJob.Status.Terminating, newJob.Status.Terminating)
+		isSuspendedWithZeroCompletions := newJob.Spec.Completions != nil && *newJob.Spec.Completions == 0 && newJob.Spec.Suspend != nil && *newJob.Spec.Suspend
```

Comment:
```suggestion
		isSuspendedWithZeroCompletions := ptr.Equal(newJob.Spec.Completions, ptr.To[int32](0)) && ptr.Equal(newJob.Spec.Suspend, ptr.To(true))
```

We can simplify this criteria by `ptr` libs.

Reply:
Done, also I swap the order to align with the variable name